lib/view_models/page_view_model/join_organization_viewModel.dart

```diff
@@ -0,0 +1,358 @@
+        variables: {'id': _currentUserId}));
+
+    if (result.hasException || userDetailsResult.hasException) {
+      print(result.exception);
+      showError(result.exception.toString());
+    } else if (!result.hasException &&
+        !disposed &&
+        !userDetailsResult.hasException) {
+      _organizationInfo = result.data['organizations'] as List;
+      _organizationInfo = result.data['organizations'] as List;
+
+      // Get the details of joined organizations.
+      joinedOrganizations =
+          userDetailsResult.data['users'][0]['joinedOrganizations'] as List;
+
+      // Get the id's of joined organizations.
+      joinedOrganizations.forEach((element) {
+        joinedOrganizationsIds.add(element['_id']);
+      });
+
+      // Filtering out organizations that are created by current user.
+      _organizationInfo = organizationInfo
+          .where((element) => element['admins'][0]['_id'] != _currentUserId)
+          .toList();
+
+      // Filtering out organizations that are already joined by user.
+      joinedOrganizationsIds.forEach((e) {
+        print(e);
+        _organizationInfo =
+            organizationInfo.where((element) => element['_id'] != e).toList();
+      });
+      notifyListeners();
+    }
+  }
+
+  Future joinPrivateOrg(BuildContext context) async {
+    //function called if the person wants to enter a private organization
+    final GraphQLClient _client = graphQLConfiguration.authClient();
+
+    final QueryResult result = await _client.mutate(MutationOptions(
+        documentNode: gql(_query.sendMembershipRequest(_itemIndex))));
+
+    if (result.hasException &&
+        result.exception.toString().substring(16) == accessTokenException) {
+      print("getting new token");
+      _authController.getNewToken();
+      return joinPrivateOrg(context);
+    } else if (result.hasException &&
+        result.exception.toString().substring(16) != accessTokenException) {
+      exceptionToast(result.exception.toString().substring(16));
+    } else if (!result.hasException && !result.loading) {
+      successToast("Request Sent to Organization Admin");
+      Navigator.pop(context);
+    }
+  }
+
+  Future joinPublicOrg(BuildContext context, String orgName) async {
+    //function which will be called if the person wants to join the organization which is not private
+    final GraphQLClient _client = graphQLConfiguration.authClient();
+    print("from function :" + _itemIndex);
+    final QueryResult result = await _client.mutate(
+        MutationOptions(documentNode: gql(_query.getOrgId(_itemIndex))));
+
+    if (result.hasException &&
+        result.exception.toString().substring(16) == accessTokenException) {
+      _authController.getNewToken();
+      return joinPublicOrg(context, orgName);
+      exceptionToast(result.exception.toString().substring(16));
+    } else if (result.hasException &&
+        result.exception.toString().substring(16) != accessTokenException) {
```

Comment:
We can do not checks with the previous comment I made.

Reply:
Sir, please check the latest commit, I've added the retrieve function in it.